### PR TITLE
[configure] Move test-suite configure parts to test-suite/

### PR DESCRIPTION
**Pull request by ejgallego**
This removes the `config/Makefile` file, and paves the way for further cleanups here; hopefully the test suite can soon run on dune or some other platform that can provide the info now present in `test-suite/test_suite_config.inc`.

This also makes the test suite Makefile to depend on less files outside `test-suite`.

cc #7149 https://github.com/coq/coq/pull/18284 https://github.com/coq/coq/pull/18287

Dependency: waiting for #18284

### CODE REVIEW

File: test-suite/Makefile [hunk 1]
```diff
@@ -262,6 +247,24 @@ summary.log:
...
+
+# EGJA: using $(ROOT)/_build/default/... here confuses make due to quotes...
+ifeq ($(INSIDE_DUNE),)
+TESTCONFIGURE:=../_build/default/test-suite/tools/coq_config_to_make.exe
+else
+TESTCONFIGURE:=./tools/coq_config_to_make.exe
+endif
+
+$(TESTCONFIGURE): tools/coq_config_to_make.ml
+	dune build ./tools/coq_config_to_make.exe
```
Comment (SkySkimmer) [inline test-suite/Makefile:262]:
```suggestion
ifeq ($(INSIDE_DUNE),)
TESTCONFIGURE:=../_build/default/test-suite/tools/coq_config_to_make.exe
else
TESTCONFIGURE:=./tools/coq_config_to_make.exe

$(TESTCONFIGURE): tools/coq_config_to_make.ml
	dune build ./tools/coq_config_to_make.exe
endif
```
we shouldn't need to build it from inside dune IIUC

File: test-suite/Makefile [hunk 2]
```diff
@@ -146,6 +140,7 @@ VSUBSYSTEMS := prerequisite success failure bugs bugs-nocoqchk output output-coq
 # All subsystems
 SUBSYSTEMS := $(VSUBSYSTEMS) misc ide ide vio coqchk output-coqchk coqwc coq-makefile tools $(UNIT_TESTS)
 
+# EJGA: This seems dangerous as first target...
```
Comment (SkySkimmer) [inline test-suite/Makefile:143]: we should just put `.DEFAULT_GOAL := all` somewhere (end or beginning of file) as we do in CoqMakefile.in

**Review by SkySkimmer — APPROVED** [review]
The principle looks sound, I think we can improve a bit how dune calls the makefile though (cf comments)

**Comment by SkySkimmer** [comment]
also we need to stop using -warn-error in the macos CI script

File: test-suite/Makefile (hunk 1 above)
Comment (ejgallego) [reply test-suite/Makefile:262]: Yes, we can do that if you prefer. Indeed there are many options here, happy to follow the last one.

File: test-suite/Makefile (hunk 2 above)
Comment (ejgallego) [reply test-suite/Makefile:143]: I did try that in the past, however I was not able to make it work lol

File: test-suite/Makefile (hunk 1 above)
Comment (SkySkimmer) [reply test-suite/Makefile:262]: Yes, depending on the .inc directly makes the most sense to me

Comment (ejgallego) [reply test-suite/Makefile:262]: Done, still we have a bit of nuissance here if the file is in tree, dune will complain about having the built-in rule and our rule generating the file.

**Comment by ejgallego** [comment]
> also we need to stop using -warn-error in the macos CI script

Done, I've kept the argument parsing in `configure` for now, but made it emit a warning.

I've also added some usability improvements for the rules, still it is needed to do a `make -C test-suite clean` before using the dune version.

I don't think we need to expose the preference for warn_error in the coq_config_make script.

File: test-suite/Makefile (hunk 1 above)
Comment (ejgallego) [reply test-suite/Makefile]: Yes, tho I am generating this file now in _build, so the problem is less.

Anyways IMVVHO we should finish Ali PR, we can get together one day or something. Even if not perfect, IMO the model is much better.

**Comment by SkySkimmer** [comment]
#18284 is merged

**Comment by ejgallego** [comment]
@coqbot run full CI

File: test-suite/Makefile [hunk 3]
```diff
@@ -260,6 +253,22 @@ summary.log:
...
+# Not inside Dune, build by hand
+ifeq ($(INSIDE_DUNE),)
+# EGJA: using $(ROOT)/_build/default/... here confuses make due to quotes...
+TESTCONFIGURE:=../_build/default/test-suite/tools/coq_config_to_make.exe
+$(TESTCONFIGURE): tools/coq_config_to_make.ml
+	dune build ./tools/coq_config_to_make.exe
+
+$(TEST_SUITE_CONFIG): $(TESTCONFIGURE)
+	chmod u+w $@ || true # Be more user-friendly w.r.t. Dune cache read-only stuff
+	$(TESTCONFIGURE) > $@
```
Comment (SkySkimmer) [inline test-suite/Makefile:269]: Would 
```suggestion
$(TEST_SUITE_CONFIG): tools/coq_config_to_make.ml
	dune build $@
```
work here? then we don't fight the dune cache

Comment (SkySkimmer) [reply test-suite/Makefile:269]: ping @ejgallego

Comment (SkySkimmer) [reply test-suite/Makefile:269]: ping @ejgallego

Comment (ejgallego) [reply test-suite/Makefile:269]: Had to tweak the variables slightly, as the command has to be the basename, but when called directly from the makefile we need to preprend the `_build` path, but indeed, that seems like a good idea, and seems to work too, thanks.

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**